lib/iris/tests/unit/analysis/cartography/test__get_lon_lat_coords.py

```diff
@@ -0,0 +1,79 @@
+# Copyright Iris contributors
+#
+# This file is part of Iris and is released under the LGPL license.
+# See COPYING and COPYING.LESSER in the root of the repository for full
+# licensing details.
+"""Test function :func:`iris.analysis.cartography._get_lon_lat_coords"""
+
+import pytest
+
+from iris.analysis.cartography import _get_lon_lat_coords as g_lat_lon
+from iris.coords import AuxCoord
+from iris.tests.stock import lat_lon_cube
+
+dim_only_cube = lat_lon_cube()
+lon_dim = dim_only_cube.coord("longitude")
+lat_dim = dim_only_cube.coord("latitude")
+
+
+def test_dim_only():
+    lon, lat = g_lat_lon(dim_only_cube)
+    assert lon == lon_dim
+    assert lat == lat_dim
+
+
+dim_aux_cube = lat_lon_cube()
+lat_aux = AuxCoord.from_coord(lat_dim)
+lat_aux.standard_name = "grid_latitude"
+lon_aux = AuxCoord.from_coord(lon_dim)
+lon_aux.standard_name = "grid_longitude"
+dim_aux_cube.add_aux_coord(lat_aux, 0)
+dim_aux_cube.add_aux_coord(lon_aux, 1)
+
+
+def test_dim_and_aux():
+    lon, lat = g_lat_lon(dim_aux_cube)
+    assert lon == lon_dim
+    assert lat == lat_dim
+
+
+aux_only_cube = dim_aux_cube.copy(dim_aux_cube.data)
```

Comment:
I'm uneasy with the pattern of re-using one object between tests (here, dim_aux_cube.data). I can appreciate that this could reduce memory use, though there still seems to be an inconsistency between copying and calling `lat_lon_cube()`